design-proposals/emts-scale-provisioning.md

```diff
@@ -0,0 +1,321 @@
+
+This approach will use the existing EIM standalone profile. This is an EMF deployment profile that only deploys EIM with Observability and FPS components.
+
+#### Option 2: A minimal EIM deployment
+
+Even the slimmed down version of EIM from Option 1 may not meet requirements for fast and easy deployment on-site.
+
+This option is a radical approach to make EIM super-lightweight and reduce any FPS dependencies that are not essential to EIM. 
+Note that this option requires some modifications to EIM components, as some of them rely on FPS components (e.g., Vault or Keylocak).
+
+The minimal EIM deployment is mainly used to support OXM use cases, where EIM is only used to provision ENs that are further shipped to the target location.
+
+The minimal EIM deployment at high level:
+- No cluster and application orchestrator deployed.
+- Simplified deployment with just Helm on a standard K8s cluster. No ArgoCD, Fleet, RKE2.
+- A single tenant deployment, without the Multi-Tenancy/IAM framework. A default tenant will be created to "own" EIM resources.
+- Reduced EIM with infra-core (without exporter, Tenant Controller), infra-onboarding and infra-managers
+- Minimal set of FPS services. A rough minimum of FPS services that may be required by EIM are:
+  - Postgres database
+  - RS proxy
+  - Vault (?)
+  - Traefik
+  - cert-manager
+  - MetalLB
+- Relaxed security requirements:
+  - No Istio for inter-Pod communication security
+  - No Kyverno for policies
+- No UI deployed, the EN pre-registration is done via Bulk Import Tool, API or CLI tool.
+- Limited observability stack that should only provide logs. No alerting, SRE, Prometheus.
+- Abandon HA requirements - EIM-S is primarily a single-node, on-prem deployment. Any data loss can be backed by hardware-level redundancy. Think of EIM-S as Rancher Desktop or Virtualbox.
+
+The minimal EIM deployment should consist of:
+- infra-core without exporter and 
+
+### Rationale
+
+The current design proposal allows to easily support legacy PXE boot while keeping the current UX around EN pre-registration. It also lets us keep all the features that EIM currently supports,
+including logging, KPI instrumentation, observability, etc.
+
+The alternative considered was to use a standalone Tinkerbell deployment without the rest of the EIM stack. While it has advantage of a more lightweight deployment,
+it would completely change the UX as we would need to familiarize customers with Tinkerbell APIs (or have a custom CLI tool to help them manage Tinkerbell CRDs).
+With the current proposal we keep using the current UX, with possibility to use Bulk Import Tool to scale preregistration process and selectively choose OS profiles for Edge Nodes.
+
+## Affected components and Teams
+
+**FPS team** needs provide help for the following:
+- Reduce FPS components to minimum to make EIM-S deployment lightweight
+- Make EIM-S deployment operation easy in terms of number of CLI commands/steps to execute
```

Comment:
I would also add "fast", for example checking if we can parallelize anything that we currently don't parallelize.